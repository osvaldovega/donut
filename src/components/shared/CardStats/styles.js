import { StyleSheet } from 'react-native';
import Colors from '../../../constants/Colors';

const styles = StyleSheet.create({
  wrapper: {
    width: '100%',
    height: 'auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: Colors.picoVoid,
    borderRadius: 6,
    borderWidth: 1,
    borderColor: Colors.white,
    padding: 10,
    marginTop: 10,
  },
  titleContainer: {
    width: '100%',
    height: 50,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  detailsContainer: {
    width: '100%',
    height: 55,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    borderTopColor: Colors.antiFlashWhite,
    borderTopWidth: 1,
  },
  detailsColumn: {
    width: '50%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  columnDivider: {
    borderLeftWidth: 1,
    borderLeftColor: Colors.antiFlashWhite,
  },
});

export default styles;
