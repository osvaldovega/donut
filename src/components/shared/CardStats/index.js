import React from 'react';
import { View } from 'react-native';
import TextHeader from '../TextHeader';
import TextLabel from '../TextLabel';
import TextContent from '../TextContent';
import styles from './styles';

const CardStats = ({
  high,
  id,
  last,
  low,
  open,
  volume_30day,
  volume,
}) => (
  <View style={styles.wrapper}>
    <View style={styles.titleContainer}>
      <TextHeader title={id} />
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <TextLabel content={'Open'} />
        <TextContent content={open} />
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <TextLabel content={'last'} />
        <TextContent content={last} />
      </View>
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <TextLabel content={'High'} />
        <TextContent content={high} />
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <TextLabel content={'Low'} />
        <TextContent content={low} />
      </View>
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <TextLabel content={'Volume'} />
        <TextContent content={volume} />
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <TextLabel content={'Volumen 30 day'} />
        <TextContent content={volume_30day} />
      </View>
    </View>
  </View>
);

CardStats.defaultProps = {
  hig: 0,
  id: 0,
  last: 0,
  low: 0,
  open: 0,
  volume_30day: 0,
  volume: 0,
}

export default CardStats;
