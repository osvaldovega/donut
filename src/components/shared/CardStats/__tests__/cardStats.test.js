import React from 'react';
import renderer from 'react-test-renderer';
import CardStats from '../index';

describe('<CardStats />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(<CardStats />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
