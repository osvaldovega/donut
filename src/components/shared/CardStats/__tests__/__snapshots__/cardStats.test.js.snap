// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardStats /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#192a56",
      "borderColor": "#ffffff",
      "borderRadius": 6,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "flex-start",
      "marginTop": 10,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "100%",
          },
          "",
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#44bd32",
              "fontFamily": "montserrat",
              "fontSize": 18,
              "fontWeight": "600",
            },
            "",
          ]
        }
      >
        0
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#f1f2f6",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Open
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          Object {
            "borderLeftColor": "#f1f2f6",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          last
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#f1f2f6",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          High
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          Object {
            "borderLeftColor": "#f1f2f6",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Low
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#f1f2f6",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Volume
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          Object {
            "borderLeftColor": "#f1f2f6",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Volumen 30 day
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
</View>
`;
